src/Features/CSharp/Portable/MakeMemberStatic/CSharpMakeMemberStaticCodeFixProvider.cs

```diff
@@ -28,8 +28,22 @@ public CSharpMakeMemberStaticCodeFixProvider()
                 "CS0708" // 'MyMethod': cannot declare instance members in a static class
             );
 
-        protected override bool IsValidMemberNode(SyntaxNode node) =>
-            node is MemberDeclarationSyntax ||
-            (node.IsKind(SyntaxKind.VariableDeclarator) && node.Ancestors().Any(a => a.IsKind(SyntaxKind.FieldDeclaration) || a.IsKind(SyntaxKind.EventFieldDeclaration)));
+        protected override bool TryGetMemberDeclaration(SyntaxNode node, [NotNullWhen(true)] out SyntaxNode? memberDeclaration)
+        {
+            if (node is MemberDeclarationSyntax)
+            {
+                memberDeclaration = node;
+                return true;
+            }
+
+            if (node.IsKind(SyntaxKind.VariableDeclarator))
+            {
+                memberDeclaration = node.FirstAncestorOrSelf<MemberDeclarationSyntax>(a => a.IsKind(SyntaxKind.FieldDeclaration) || a.IsKind(SyntaxKind.EventFieldDeclaration));
```

Comment:
instead of walking arbitrarily high, we should check for exactly the shapes we expect here.  that prevents thigns like matching on a variable inside a lambda inside a field initializer.  In this case, we jsut want to match `node { Parent: VaraibleDeclaration { Parent: FieldDeclarationSyntax or EventFieldDeclarationSyntax } }`